solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -271,17 +287,23 @@ public void transform(SolrDocument doc, int docid)
 
     private void implTransform(SolrDocument doc, int docid, Float score)
         throws IOException {
-      Object fv = featureLogger.getFeatureVector(docid, scoringQuery, searcher);
-      if (fv == null) { // FV for this document was not in the cache
-        fv = featureLogger.makeFeatureVector(
-            LTRRescorer.extractFeaturesInfo(
-                modelWeight,
-                docid,
-                (docsWereNotReranked ? score : null),
-                leafContexts));
+      LTRScoringQuery rerankingQuery = rerankingQueries[0];
+      LTRScoringQuery.ModelWeight rerankingModelWeight = modelWeights[0];
+      if (rerankingQueries.length > 1 && rerankingQueries[1].getPickedInterleavingDocIds().contains(docid)) {
+        rerankingQuery = rerankingQueries[1];
```

Comment:
5/n When doing the 4/n stuff a question occurred to me here: `rerankingQueries[0]` and `modelWeights[0]` are the initial values. If the query is changed to `rerankingQueries[1]` then should the weight also be changed to `modelWeights[1]`? (Open question, haven't analysed it further yet nor tried it out.)

Reply:
Yes, I agree, I'll change that and resolve this later